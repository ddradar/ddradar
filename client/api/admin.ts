import type { NuxtHTTPInstance } from '@nuxt/http'

import type { SongInfo } from '~/api/song'
import type { NotificationSchema } from '~/core/db/notification'

type NotificationRequest = Omit<
  NotificationSchema,
  'id' | 'sender' | 'timeStamp'
> & {
  id?: string
  timeStamp?: number
}

/**
 * Call "Post Notification" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/postNotification
 */
export function postNotification(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: NotificationRequest
) {
  return $http.$post<NotificationRequest>('/api/v1/admin/notification', {
    ...body,
    sender: 'SYSTEM',
  })
}

/**
 * Call "Post Song Information" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/postSongInfo
 */
export function postSongInfo(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: SongInfo
) {
  return $http.$post<SongInfo>('/api/v1/admin/songs', body)
}
