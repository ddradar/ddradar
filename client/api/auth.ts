import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

type AuthResult = {
  clientPrincipal: Api.ClientPrincipal | null
}

/**
 * Call "Get Credentials" API.
 * @see https://docs.microsoft.com/azure/static-web-apps/user-information
 */
export async function getClientPrincipal(
  $http: Pick<NuxtHTTPInstance, '$get'>
) {
  const { clientPrincipal } = await $http.$get<AuthResult>('/.auth/me')
  return clientPrincipal
}
