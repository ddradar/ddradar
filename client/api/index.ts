export const apiPrefix = '/api/v1'
