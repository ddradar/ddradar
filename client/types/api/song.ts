/**
 * Object type returned by `/api/v1/songs/{:songId}`
 * @see https://github.com/ddradar/ddradar/blob/master/api/getSongInfo/README.md
 */
export type SongInfo = {
  /**
   * Song id that depend on official site.
   * @example `^([01689bdiloqDIOPQ]*){32}$`
   */
  id: string
  name: string
  /**
   * Song furigana for sorting.
   * @example `^([A-Z0-9 .ぁ-んー]*)$`
   */
  nameKana: string
  /**
   * Index for sorting. Associated with the "Choose by Name" folder.
   * @example `0`: あ行, `1`: か行, ..., `10`: A, `11`: B, ..., `35`: Z, `36`: 数字・記号
   */
  nameIndex: number
  artist: string
  /** Series title depend on official site. */
  series: string
  /**
   * Displayed min BPM (Beet Per Minutes).
   * Set to `null` if not revealed, such as "???".
   */
  minBPM: number | null
  /**
   * Displayed max BPM (Beet Per Minutes).
   * Set to `null` if not revealed, such as "???".
   */
  maxBPM: number | null
  charts: StepChart[]
}

export type StepChart = {
  /** `1`: SINGLE, `2`: DOUBLE */
  playStyle: 1 | 2
  /**
   * `0`: BEGINNER,
   * `1`: BASIC,
   * `2`: DIFFICULT,
   * `3`: EXPERT,
   * `4`: CHALLENGE
   */
  difficulty: 0 | 1 | 2 | 3 | 4
  level: number
  /** Normal arrow count. (Jump = 1 count) */
  notes: number
  /** Freeze Arrow count. */
  freezeArrow: number
  /** Shock Arrow count. */
  shockArrow: number
  /** Groove Radar STREAM */
  stream: number
  /** Groove Radar VOLTAGE */
  voltage: number
  /** Groove Radar AIR */
  air: number
  /** Groove Radar FREEZE */
  freeze: number
  /** Groove Radar CHAOS */
  chaos: number
}

/**
 * Object type returned by `/api/v1/songs/name/{:name}` and `/api/v1/songs/series/{:series}`
 * @see https://github.com/ddradar/ddradar/blob/master/api/searchSongByName/README.md
 * @see https://github.com/ddradar/ddradar/blob/master/api/searchSongBySeries/README.md
 */
export type SongListData = Omit<SongInfo, 'charts'>

/**
 * Object type returned by `/api/v1/charts/{:playStyle}/{:level}`
 * @see https://github.com/ddradar/ddradar/blob/master/api/searchCharts/README.md
 */
export type ChartInfo = Pick<SongInfo, 'id' | 'name' | 'series'> &
  Pick<StepChart, 'playStyle' | 'difficulty' | 'level'>

export const SeriesList: string[] = [
  'DDR 1st',
  'DDR 2ndMIX',
  'DDR 3rdMIX',
  'DDR 4thMIX',
  'DDR 5thMIX',
  'DDRMAX',
  'DDRMAX2',
  'DDR EXTREME',
  'DDR SuperNOVA',
  'DDR SuperNOVA2',
  'DDR X',
  'DDR X2',
  'DDR X3 VS 2ndMIX',
  'DanceDanceRevolution (2013)',
  'DanceDanceRevolution (2014)',
  'DanceDanceRevolution A',
  'DanceDanceRevolution A20',
  'DanceDanceRevolution A20 PLUS',
]

export const NameIndexList: string[] = [
  'あ',
  'か',
  'さ',
  'た',
  'な',
  'は',
  'ま',
  'や',
  'ら',
  'わ',
  'A',
  'B',
  'C',
  'D',
  'E',
  'F',
  'G',
  'H',
  'I',
  'J',
  'K',
  'L',
  'M',
  'N',
  'O',
  'P',
  'Q',
  'R',
  'S',
  'T',
  'U',
  'V',
  'W',
  'X',
  'Y',
  'Z',
  '数字・記号',
]

export function shortenSeriesName(series: string) {
  return series.replace(/^(DDR |DanceDanceRevolution )\(?([^)]+)\)?$/, '$2')
}

export function getPlayStyleName(
  playStyle: number
): 'SINGLE' | 'DOUBLE' | 'UNKNOWN' {
  return (['SINGLE', 'DOUBLE'] as const)[playStyle - 1] ?? 'UNKNOWN'
}

export function getDifficultyName(
  difficulty: number
): 'BEGINNER' | 'BASIC' | 'DIFFICULT' | 'EXPERT' | 'CHALLENGE' | 'UNKNOWN' {
  return (
    (['BEGINNER', 'BASIC', 'DIFFICULT', 'EXPERT', 'CHALLENGE'] as const)[
      difficulty
    ] ?? 'UNKNOWN'
  )
}
