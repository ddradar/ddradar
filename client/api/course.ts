import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

import { apiPrefix } from '~/api'

export function getCourseType(type: number) {
  return type === 1 ? 'NONSTOP' : type === 2 ? '段位認定' : ''
}

/**
 * Call "Get Course List" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/courses/
 */
export function getCourseList(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  series?: 16 | 17,
  type?: 1 | 2
) {
  const searchParams = new URLSearchParams()
  if (series) searchParams.append('series', `${series}`)
  if (type) searchParams.append('type', `${type}`)
  return $http.$get<Api.CourseListData[]>(`${apiPrefix}/courses`, {
    searchParams,
  })
}

/**
 * Call "Get Course Information" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/courses__id/
 */
export function getCourseInfo(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  id: string
) {
  return $http.$get<Api.CourseInfo>(`${apiPrefix}/courses/${id}`)
}
