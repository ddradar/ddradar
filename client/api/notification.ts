import type { Api } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

import { apiPrefix } from '~/api'

/**
 * Call "Get Notification List" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/notification/
 */
export function getNotificationList(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  topOnly?: boolean
) {
  const query = topOnly ? '?scope=top' : ''
  return $http.$get<Api.NotificationListData[]>(
    `${apiPrefix}/notification${query}`
  )
}

/**
 * Call "Get Notification Info" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/notification__id/
 */
export function getNotificationInfo(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  id: string
) {
  return $http.$get<Api.NotificationInfo>(`${apiPrefix}/notification/${id}`)
}

/**
 * Call "Post Notification" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/notification--post/
 */
export function postNotification(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: Omit<Api.NotificationBody, 'sender'>
) {
  return $http.$post<Api.NotificationBody>(`${apiPrefix}/notification`, {
    ...body,
    sender: 'SYSTEM',
  })
}
