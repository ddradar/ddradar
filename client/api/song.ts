import type { Api } from '@ddradar/core'
import { Song } from '@ddradar/core'
import type { NuxtHTTPInstance } from '@nuxt/http'

import { apiPrefix } from '~/api'

export function shortenSeriesName(series: string) {
  return series.replace(/^(DDR |DanceDanceRevolution )\(?([^)]+)\)?$/, '$2')
}

export function getChartTitle({
  playStyle,
  difficulty,
  level,
}: Pick<Api.ChartInfo, 'playStyle' | 'difficulty' | 'level'>) {
  const shortPlayStyle = `${Song.playStyleMap.get(playStyle)![0]}P` as
    | 'SP'
    | 'DP'
  const difficultyName = Song.difficultyMap.get(difficulty)!
  return `${shortPlayStyle}-${difficultyName} (${level})`
}

export function getDisplayedBPM({
  minBPM,
  maxBPM,
}: Pick<Api.SongInfo, 'minBPM' | 'maxBPM'>) {
  if (!minBPM || !maxBPM) return '???'
  if (minBPM === maxBPM) return `${minBPM}`
  return `${minBPM}-${maxBPM}`
}

/**
 * Call "Get Song Information" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/songs__id/
 */
export function getSongInfo($http: Pick<NuxtHTTPInstance, '$get'>, id: string) {
  return $http.$get<Api.SongInfo>(`${apiPrefix}/songs/${id}`)
}

/**
 * Call "Search Song by Name" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/songs__name__id/
 */
export function searchSongByName(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  nameIndex: number
) {
  return $http.$get<Api.SongListData[]>(`${apiPrefix}/songs/name/${nameIndex}`)
}

/**
 * Call "Search Song by Series" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/songs__series__id/
 */
export function searchSongBySeries(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  series: number
) {
  return $http.$get<Api.SongListData[]>(`${apiPrefix}/songs/series/${series}`)
}

/**
 * Call "Search Charts" API.
 * @see https://github.com/ddradar/ddradar/tree/master/api/charts__style__level/
 */
export function searchCharts(
  $http: Pick<NuxtHTTPInstance, '$get'>,
  playStyle: Song.PlayStyle,
  level: number
) {
  return $http.$get<Api.ChartInfo[]>(
    `${apiPrefix}/charts/${playStyle}/${level}`
  )
}

/**
 * Call "Post Song Information" API. (Admin only)
 * @see https://github.com/ddradar/ddradar/tree/master/api/songs--post/
 */
export function postSongInfo(
  $http: Pick<NuxtHTTPInstance, '$post'>,
  body: Api.SongInfo
) {
  return $http.$post<Api.SongInfo>(`${apiPrefix}/songs`, body)
}
