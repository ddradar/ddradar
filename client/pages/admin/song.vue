<template>
  <section class="section">
    <h1 class="title">Add/Update Song</h1>
    <b-field label="Song ID">
      <b-field grouped>
        <b-input
          v-model="id"
          maxlength="32"
          required
          pattern="^[01689bdiloqDIOPQ]{32}$"
        />
        <b-button
          type="is-primary"
          :disabled="!isValidSongId"
          @click="loadSongInfo()"
        >
          Load
        </b-button>
      </b-field>
    </b-field>

    <b-field label="Name">
      <b-input v-model="name" required />
    </b-field>

    <b-field label="Furigana">
      <b-input
        v-model="nameKana"
        required
        pattern="^([A-Z0-9 .ぁ-んー]*)$"
        placeholder="A-Z 0-9 . あ-ん ー"
      />
    </b-field>

    <b-field label="Artist">
      <b-input v-model="artist" />
    </b-field>

    <b-field label="BPM">
      <b-field grouped>
        <b-input v-model.number="minBPM" type="number" placeholder="min" />
        <span class="control">-</span>
        <b-input v-model.number="maxBPM" type="number" placeholder="max" />
      </b-field>
    </b-field>

    <b-field v-for="(chart, i) in charts" :key="i" :label="`Chart ${i + 1}`">
      <b-field grouped group-multiline>
        <b-select placeholder="PlayStyle"></b-select>
        <b-select placeholder="Difficulty"></b-select>
        <b-input placeholder="Level" type="number" min="1" max="20" />
        <b-field>
          <b-input placeholder="Notes" type="number" min="0" max="9999" />
          <b-input placeholder="FA" type="number" min="0" max="9999" />
          <b-input placeholder="SA" type="number" min="0" max="9999" />
        </b-field>
        <b-field>
          <b-input placeholder="Str" type="number" min="0" max="300" />
          <b-input placeholder="Vol" type="number" min="0" max="300" />
          <b-input placeholder="Air" type="number" min="0" max="300" />
          <b-input placeholder="Fre" type="number" min="0" max="300" />
          <b-input placeholder="Cha" type="number" min="0" max="300" />
        </b-field>
      </b-field>
    </b-field>
    <b-field>
      <b-button icon-left="plus">Add Chart</b-button>
    </b-field>

    <b-field>
      <b-button type="is-success">Save</b-button>
    </b-field>
  </section>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator'

import { SongInfo, StepChart } from '~/types/api/song'

@Component
export default class SongEditorPage extends Vue implements SongInfo {
  id: string = ''
  name: string = ''
  nameKana: string = ''
  nameIndex: number = 0
  artist: string = ''
  series: string = ''
  minBPM: number | null = null
  maxBPM: number | null = null
  charts: StepChart[] = [
    {
      playStyle: 1,
      difficulty: 0,
      level: 3,
      notes: 67,
      freezeArrow: 0,
      shockArrow: 0,
      stream: 14,
      voltage: 14,
      air: 9,
      freeze: 0,
      chaos: 0,
    },
    {
      playStyle: 1,
      difficulty: 1,
      level: 7,
      notes: 143,
      freezeArrow: 0,
      shockArrow: 0,
      stream: 31,
      voltage: 29,
      air: 47,
      freeze: 0,
      chaos: 3,
    },
    {
      playStyle: 1,
      difficulty: 2,
      level: 9,
      notes: 188,
      freezeArrow: 0,
      shockArrow: 0,
      stream: 41,
      voltage: 39,
      air: 27,
      freeze: 0,
      chaos: 13,
    },
    {
      playStyle: 1,
      difficulty: 3,
      level: 12,
      notes: 212,
      freezeArrow: 0,
      shockArrow: 0,
      stream: 46,
      voltage: 39,
      air: 54,
      freeze: 0,
      chaos: 19,
    },
    {
      playStyle: 2,
      difficulty: 1,
      level: 7,
      notes: 130,
      freezeArrow: 0,
      shockArrow: 0,
      stream: 28,
      voltage: 29,
      air: 61,
      freeze: 0,
      chaos: 1,
    },
    {
      playStyle: 2,
      difficulty: 2,
      level: 9,
      notes: 181,
      freezeArrow: 0,
      shockArrow: 0,
      stream: 40,
      voltage: 39,
      air: 30,
      freeze: 0,
      chaos: 11,
    },
    {
      playStyle: 2,
      difficulty: 3,
      level: 11,
      notes: 220,
      freezeArrow: 0,
      shockArrow: 0,
      stream: 48,
      voltage: 54,
      air: 27,
      freeze: 0,
      chaos: 41,
    },
  ]

  chartIndex: number = 0

  get isValidSongId() {
    return /^[01689bdiloqDIOPQ]{32}$/.test(this.id)
  }

  async loadSongInfo() {
    if (!this.isValidSongId) return
    try {
      const songInfo = await this.$http.$get<SongInfo>(`/songs/${this.id}`)

      this.name = songInfo.name
      this.nameKana = songInfo.nameKana
      this.nameIndex = songInfo.nameIndex
      this.artist = songInfo.artist
      this.series = songInfo.series
      this.minBPM = songInfo.minBPM
      this.maxBPM = songInfo.maxBPM
    } catch {}
  }
}
</script>
