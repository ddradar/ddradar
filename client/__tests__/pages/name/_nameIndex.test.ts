import type { Context } from '@nuxt/types'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import Buefy from 'buefy'
import { mocked } from 'ts-jest/utils'

import { searchSongByName } from '~/api/song'
import SongByNamePage from '~/pages/name/_nameIndex.vue'

jest.mock('~/api/song')
const localVue = createLocalVue()
localVue.use(Buefy)

describe('pages/name/_nameIndex.vue', () => {
  const $fetchState = { pending: false }
  describe('validate()', () => {
    test.each(['', 'foo', '99', '-1', '1.0'])(
      '/name/%s returns false',
      nameIndex => {
        // Arrange
        const wrapper = shallowMount(SongByNamePage, {
          localVue,
          mocks: { $route: { params: { nameIndex } }, $fetchState },
        })
        const ctx = { params: { nameIndex } } as unknown as Context

        // Act - Assert
        expect(wrapper.vm.$options.validate!(ctx)).toBe(false)
      }
    )
    test.each(['0', '1', '9', '10', '36'])(
      '/name/%s returns true',
      nameIndex => {
        // Arrange
        const wrapper = shallowMount(SongByNamePage, {
          localVue,
          mocks: { $route: { params: { nameIndex } }, $fetchState },
        })
        const ctx = { params: { nameIndex } } as unknown as Context

        // Act - Assert
        expect(wrapper.vm.$options.validate!(ctx)).toBe(true)
      }
    )
  })
  describe('fetch()', () => {
    const apiMock = mocked(searchSongByName)
    beforeEach(() => {
      apiMock.mockClear()
      apiMock.mockResolvedValue([])
    })

    test.each(['0', '1', '9', '10', '36'])(
      'calls /songs/name/%s API',
      async nameIndex => {
        // Arrange
        const $http = { $get: jest.fn() }
        const wrapper = shallowMount(SongByNamePage, {
          localVue,
          mocks: { $route: { params: { nameIndex } }, $fetchState, $http },
        })

        // Act
        // @ts-ignore
        await wrapper.vm.$options.fetch?.call(wrapper.vm)

        // Assert
        expect(apiMock).toBeCalledTimes(1)
        expect(apiMock).toBeCalledWith($http, parseInt(nameIndex, 10))
      }
    )
  })
  describe('get title()', () => {
    test.each([
      ['0', 'あ'],
      ['9', 'わ'],
      ['10', 'A'],
      ['35', 'Z'],
      ['36', '数字・記号'],
    ])('name/%s route returns %s', (nameIndex, expected) => {
      // Arrange
      const wrapper = shallowMount(SongByNamePage, {
        localVue,
        mocks: { $route: { params: { nameIndex } }, $fetchState },
      })

      // Act - Assert
      // @ts-ignore
      expect(wrapper.vm.title).toBe(expected)
    })
  })
})
