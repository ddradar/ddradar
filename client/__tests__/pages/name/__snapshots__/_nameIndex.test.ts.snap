// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/name/_nameIndex.vue renders loaded 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    数字・記号
  </h1>
   
  <div
    class="b-table"
  >
    <!---->
     
    <!---->
     
    <div
      class="table-wrapper"
    >
      <table
        class="table is-striped"
      >
        <!---->
         
        <tbody>
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          .59
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        dj TAKA
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        135
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          1998
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        NAOKI
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        150
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          30 Lives (Up-Up-Down-Dance Mix)
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        The Motion Sick
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        144
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          321STARS
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        DJ SIMON
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        192
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          39
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        sasakure.UK×DECO*27
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        175
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          50th Memorial Songs -Beginning Story-
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        BEMANI Sound Team
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        135
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          50th Memorial Songs -Flagship medley-
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        BEMANI Sound Team
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        135-175
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          50th Memorial Songs -The BEMANI History-
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        BEMANI Sound Team
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        136
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          50th Memorial Songs -二人の時 ～under the cherry blossoms～-
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        BEMANI Sound Team
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        160
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          8000000
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        kors k
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        100-200
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          888
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        DJ TECHNORCH
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        111-888
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
          <tr
            class=""
            draggable="false"
          >
            <!---->
             
            <!---->
             
            <td
              class=""
              data-label="Name"
            >
              <a>
                
          IX
        
              </a>
            </td>
             
            <td
              class=""
              data-label="Artist"
            >
              
        dj TAKA VS DJ TOTTO feat.藍
      
            </td>
             
            <td
              class=""
              data-label="BPM"
            >
              
        99-396
      
            </td>
             
            <!---->
          </tr>
           
          <!---->
           
          <!---->
        </tbody>
         
        <!---->
      </table>
    </div>
     
    <div
      class="level"
    >
      <div
        class="level-left"
      />
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <nav
            class="pagination"
          >
            <a
              class="pagination-link pagination-previous"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-left mdi-24px"
                />
              </span>
            </a>
             
            <a
              class="pagination-link pagination-next"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-right mdi-24px"
                />
              </span>
            </a>
             
            <ul
              class="pagination-list"
            >
              <!---->
               
              <!---->
               
              <li>
                <a
                  aria-current="true"
                  class="pagination-link is-current "
                  href="#"
                  role="button"
                >
                  1
                </a>
              </li>
               
              <!---->
               
              <!---->
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`/name/_nameIndex.vue renders loading 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    数字・記号
  </h1>
   
  <div
    class="b-table is-loading"
  >
    <!---->
     
    <!---->
     
    <div
      class="table-wrapper"
    >
      <table
        class="table is-striped"
      >
        <!---->
         
        <tbody>
          <tr
            class="is-empty"
          >
            <td
              colspan="0"
            >
              <section
                class="section"
              >
                <div
                  class="content has-text-grey has-text-centered"
                >
                  <p>
                    Nothing here.
                  </p>
                </div>
              </section>
            </td>
          </tr>
        </tbody>
         
        <!---->
      </table>
    </div>
     
    <div
      class="level"
    >
      <div
        class="level-left"
      />
       
      <div
        class="level-right"
      >
        <div
          class="level-item"
        >
          <nav
            class="pagination"
          >
            <a
              class="pagination-link pagination-previous"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-left mdi-24px"
                />
              </span>
            </a>
             
            <a
              class="pagination-link pagination-next"
              disabled="disabled"
              href="#"
              role="button"
            >
              <span
                aria-hidden="true"
                class="icon"
              >
                <i
                  class="mdi mdi-chevron-right mdi-24px"
                />
              </span>
            </a>
             
            <ul
              class="pagination-list"
            >
              <!---->
               
              <!---->
                
              <!---->
               
              <!---->
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</section>
`;
